From 3edfc9b9d9377180e8c7a1a9cab3e6407c5feb10 Mon Sep 17 00:00:00 2001
Message-ID: <3edfc9b9d9377180e8c7a1a9cab3e6407c5feb10.1699195588.git.sharlatanus@gmail.com>
In-Reply-To: <1c3e20f81479ed702270fc6640939db165bd33bc.1699195588.git.sharlatanus@gmail.com>
References: <1c3e20f81479ed702270fc6640939db165bd33bc.1699195588.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 2 Nov 2023 15:13:04 +0000
Subject: [PATCH v3 5/5] gnu: elemental: Update package style.

* gnu/packages/maths.scm (elemental): Update package style, shift order
of package fields.
[arguments]: Use G-expressions. {phases}: Remove trailing #f from each
lambda.

Change-Id: Ib961eb99159e1206963041bd7014dca032a21441
---
 gnu/packages/maths.scm | 48 +++++++++++++++++++++---------------------
 1 file changed, 24 insertions(+), 24 deletions(-)

diff --git a/gnu/packages/maths.scm b/gnu/packages/maths.scm
index 0ed25834f6..34710fc65c 100644
--- a/gnu/packages/maths.scm
+++ b/gnu/packages/maths.scm
@@ -7611,7 +7611,29 @@ (define-public elemental
                  (base32
                   "06xcs4ic60ndcf2hq19gr8yjwdsnphpcyhapab41rkw726z4lm7p"))))
       (build-system cmake-build-system)
-      (home-page "https://github.com/elemental/Elemental")
+      (arguments
+       (list
+        #:build-type "Release" ;default RelWithDebInfo not supported
+        #:configure-flags
+        #~(list "-DEL_DISABLE_PARMETIS:BOOL=YES"
+                "-DEL_AVOID_COMPLEX_MPI:BOOL=NO"
+                "-DEL_CACHE_WARNINGS:BOOL=YES"
+                "-DEL_TESTS:BOOL=YES"
+                "-DCMAKE_INSTALL_LIBDIR=lib"
+                "-DGFORTRAN_LIB=gfortran")
+        #:phases
+        #~(modify-phases %standard-phases
+            (add-before 'check 'mpi-setup
+                 #$%openmpi-setup)
+            (add-before 'check 'setup-tests
+              (lambda _                ;; Parallelism is done at the MPI layer.
+                (setenv "OMP_NUM_THREADS" "1")))
+            (add-after 'install 'remove-tests
+              (lambda _
+                ;; Tests are installed, with no easy configuration
+                ;; switch to prevent this, so delete them.
+                (delete-file-recursively
+                 (string-append  #$output "/bin/test")))))))
       (native-inputs
        (list gfortran))
       (inputs
@@ -7624,29 +7646,7 @@ (define-public elemental
              openmpi
              qd
              openblas))
-      (arguments
-       `(#:build-type "Release"           ;default RelWithDebInfo not supported
-         #:configure-flags `("-DEL_DISABLE_PARMETIS:BOOL=YES"
-                             "-DEL_AVOID_COMPLEX_MPI:BOOL=NO"
-                             "-DEL_CACHE_WARNINGS:BOOL=YES"
-                             "-DEL_TESTS:BOOL=YES"
-                             "-DCMAKE_INSTALL_LIBDIR=lib"
-                             "-DGFORTRAN_LIB=gfortran")
-         #:phases (modify-phases %standard-phases
-                    (add-before 'check 'mpi-setup
-                      ,%openmpi-setup)
-                    (add-before 'check 'setup-tests
-                      (lambda _
-                        ;; Parallelism is done at the MPI layer.
-                        (setenv "OMP_NUM_THREADS" "1")
-                        #t))
-                    (add-after 'install 'remove-tests
-                      (lambda* (#:key outputs #:allow-other-keys)
-                        ;; Tests are installed, with no easy configuration
-                        ;; switch to prevent this, so delete them.
-                        (delete-file-recursively
-                         (string-append (assoc-ref outputs "out") "/bin"))
-                        #t)))))
+      (home-page "https://github.com/elemental/Elemental")
       (synopsis "Dense and sparse-direct linear algebra and optimization")
       (description "Elemental is a modern C++ library for distributed-memory
 dense and sparse-direct linear algebra, conic optimization, and lattice
-- 
2.41.0

